### PR TITLE
Import from C to C++ suspended

### PR DESCRIPTION
**Pull request by baloghadamsoftware**

### CODE REVIEW

**Comment by baloghadamsoftware** [comment]
Issue: #295

**Comment by Xazax-hun** [comment]
Hmm, I think we should also forbid the other way around, importing from C++ to C. What do you think?

File: lib/Tooling/CrossTranslationUnit.cpp [hunk 1]
```diff
@@ -235,6 +235,16 @@ const FunctionDecl *CrossTranslationUnit::getCrossTUDefinition(
     } else {
       Unit = ASTCacheEntry->second.get();
     }
+
+    const auto& LangTo = Context.getLangOpts();
+    const auto& LangFrom = Unit->getASTContext().getLangOpts();
+    // FIXME: Currenty we do not support the import of C AST into C++. This
+    // limitation should be lifted in the future after carafully examining
```
Comment (martong) [inline lib/Tooling/CrossTranslationUnit.cpp]: carafully -> carefully

**Comment by martong** [comment]
@Xazax-hun , i agree that would be the safest for now.

**Comment by dkrupp** [comment]
Adam is sick, so @bruntib it would be great to take over this patch and fix the review comments. Would be so great if it could be merged in today.

File: lib/Tooling/CrossTranslationUnit.cpp [hunk 2]
```diff
@@ -235,6 +235,17 @@ const FunctionDecl *CrossTranslationUnit::getCrossTUDefinition(
...
       Unit = ASTCacheEntry->second.get();
     }
+
+    const auto& LangTo = Context.getLangOpts();
+    const auto& LangFrom = Unit->getASTContext().getLangOpts();
+    // FIXME: Currenty we do not support the import of C AST into C++ or vica
+    // versa. This limitation should be lifted in the future after carefully
+    // examining and handling the cases where the two ASTs are incomatible.
+    if ((LangTo.CPlusPlus && !LangFrom.CPlusPlus) ||
+        ((LangTo.C99 || LangTo.C11) && !(LangFrom.C99 || LangFrom.C11))) {
```
Comment (martong) [inline lib/Tooling/CrossTranslationUnit.cpp]: Does this handle C89?
`BENIGN_LANGOPT(ImplicitInt, 1, !C99 && !CPlusPlus, "C89 implicit 'int'")`
Also, do we want to enable import from c89 to c11?

Comment (Xazax-hun) [reply lib/Tooling/CrossTranslationUnit.cpp]: Do we need this convoluted logic?
I think C11 implies C99 and so on, so each time C11 is true, C99 should be also true.
I was thinking about something like: 
```
if (LangTo.CPlusPlus != LangFrom.CPlusPlus)
```

**Pull request merged**